2-mandatory/1-weather-report.js

```diff
@@ -9,17 +9,25 @@
         - return an array of strings, which is a statement about the temperature of each city.
             For example, "The temperature in London is 10 degrees"
         - Hint: you can call the temperatureService function from your function
+        npm test -- --testPathPattern 2-logic-error
 */
 
 function getTemperatureReport(cities) {
     // TODO
-}
+    let myArray = [];
+    let message = "";
+    for (const city of cities) {
+        message = `The temperature in ${city} is ${temperatureService(city)} degrees`;
```

Comment:
Nothing wrong with the code, but I would have used .push method for the array as shown:
```suggestion
        myArray.push(`The temperature in ${city} is ${temperatureService(city)} degrees`);
```
Functionally, they do the same thing but it means you don't need additional variable `message`

Reply:
I agree with @berkeli. However, if you wanted to keep the `message` variable, you could move it inside the for loop, since it's not need outside.